c/jwk.c

```diff
@@ -0,0 +1,398 @@
+    *statusOut = JWK_STATUS_UNRECOGNIZED_FMT_ERROR;
+    return;
+  }
+
+  char exponent[strlen(exponentBase64Url)+1];
+  int exponentLen = decodeBase64Url(exponentBase64Url, exponent, &status);
+  if (status != 0) {
+    zowelog(NULL, LOG_COMP_ID_JWK, ZOWE_LOG_DEBUG, "failed to decode exponent '%s'\n", exponentBase64Url);
+    *statusOut = JWK_STATUS_UNRECOGNIZED_FMT_ERROR;
+    return;
+  }
+
+  gsk_buffer modulusBuf = { .data = (void*)modulus, .length = modulusLen };
+  gsk_buffer exponentBuf = { .data = (void*)exponent, .length = exponentLen };
+  status = gsk_construct_public_key_rsa(&modulusBuf, &exponentBuf, publicKeyOut);
+  if (status != 0) {
+    zowelog(NULL, LOG_COMP_ID_JWK, ZOWE_LOG_DEBUG, "failed to construct public key: %s\n", gsk_strerror(status));
+    *statusOut = JWK_STATUS_PUBLIC_KEY_ERROR;
+    return;
+  }
+
+  *statusOut = JWK_STATUS_OK;
+}
+
+static int decodeBase64Url(const char *data, char *resultBuf, int *statusOut) {
+  size_t dataLen = strlen(data);
+  size_t base64Size = dataLen + 16;
+  char base64[base64Size];
+
+  strcpy(base64, data);
+  if (base64urlToBase64(base64, base64Size) < 0) {
+    *statusOut = JWK_STATUS_INVALID_BASE64_URL;
+    return 0;
+  }
+
+  int decodedLen = decodeBase64(base64, resultBuf);
+  if (decodedLen <= 0) {
+    *statusOut = JWK_STATUS_INVALID_BASE64;
+    return 0;    
+  }
+  return decodedLen;
+}
+
+static int checkJwtSignature(JwsAlgorithm algorithm,
+                      int sigLen, const uint8_t *signature,
+                      int msgLen, const uint8_t *message,
+                      void *userData) {
+  JwkContext *context = userData;
+  if (!context->isPublicKeyInitialized) {
+    return RC_JWT_NOT_CONFIGURED;
+  }
+
+  if (algorithm == JWS_ALGORITHM_none) {
+    if (sigLen == 0) {
+      return RC_JWT_INSECURE;
+    } else {
+      return RC_JWT_INVALID_SIGLEN;
+    }
+  }
+
+  x509_algorithm_type alg = x509_alg_unknown;
+  gsk_buffer msgBuffer = { .length = msgLen, .data = (void*)message };
+  gsk_buffer sigBuffer = { .length = sigLen, .data = (void*)signature };
+
+  switch (algorithm) {
+    case JWS_ALGORITHM_RS256:
+      alg = x509_alg_sha256WithRsaEncryption;
+      break;
+  }
+  int gskStatus = gsk_verify_data_signature(alg, &context->publicKey, 0, &msgBuffer, &sigBuffer);
+  if (gskStatus != 0) {
+    zowelog(NULL, LOG_COMP_ID_JWK, ZOWE_LOG_DEBUG, "failed to verify signature with status %d - %s\n",
+            gskStatus, gsk_strerror(gskStatus));
+    return RC_JWT_SIG_MISMATCH;
+  }
+  return RC_JWT_OK;
+}
+
+static const char *MESSAGES[] = {
+  [JWK_STATUS_OK] = "OK",
+  [JWK_STATUS_HTTP_ERROR] = "HTTP error",
+  [JWK_STATUS_RESPONSE_ERROR] = "HTTP response error",
+  [JWK_STATUS_JSON_RESPONSE_ERROR] = "HTTP response error - invalid JSON",
+  [JWK_STATUS_UNRECOGNIZED_FMT_ERROR] = "JWK is in unrecognized format",
+  [JWK_STATUS_NOT_RSA_KEY] = "not RSA public key",
```

Comment:
Did I miss something or is this unused?

Reply:
Thanks, it was unused, I removed that status and other unused ones.